run/SU2_preCICE_CHT.py

```diff
@@ -26,240 +26,247 @@
+
+	# If the CHT marker is defined on this rank:
+	if CHTMarkerID != None:
+		nVertex_CHTMarker = SU2Driver.GetNumberVertices(CHTMarkerID) #Total number of vertices on the marker
+		nVertex_CHTMarker_HALO = SU2Driver.GetNumberHaloVertices(CHTMarkerID)
+		nVertex_CHTMarker_PHYS = nVertex_CHTMarker - nVertex_CHTMarker_HALO
+
+		# Obtain indices of all vertices that are being worked on on this rank
+		for iVertex in range(nVertex_CHTMarker):
+			if not SU2Driver.IsAHaloNode(CHTMarkerID, iVertex):
+				iVertices_CHTMarker_PHYS.append(int(iVertex))
+
+	# Get coords of vertices
+	coords = numpy.zeros((nVertex_CHTMarker_PHYS, options.nDim))
+	for i, iVertex in enumerate(iVertices_CHTMarker_PHYS):
+		coord_passive = SU2Driver.GetInitialMeshCoord(CHTMarkerID, iVertex)
+		for iDim in range(options.nDim):
+			coords[i, iDim] = coord_passive[iDim]
+
+	# Set mesh vertices in preCICE:
+	try:
+		vertex_ids = participant.set_mesh_vertices(mesh_name, coords)
+	except:
+		print("Could not set mesh vertices for preCICE. Was a (known) mesh specified in the options?")
+		return
+
+	# Get read and write data IDs
+	precice_read = "Temperature"
+	precice_write = "Heat-Flux"
+	GetFxn = lambda *args: -1*SU2Driver.GetVertexNormalHeatFlux(*args)
+	SetFxn = SU2Driver.SetVertexTemperature
+	GetInitialFxn = SU2Driver.GetVertexTemperature
+	# Reverse coupling data read/write if -r flag included
+	if options.precice_reverse:
+		precice_read = "Heat-Flux"
+		precice_write = "Temperature"
+		GetFxn = SU2Driver.GetVertexTemperature
+		SetFxn = SU2Driver.SetVertexNormalHeatFlux
+		GetInitialFxn = lambda *args: -1*SU2Driver.GetVertexNormalHeatFlux(*args)
+
+	# Instantiate arrays to hold temperature + heat flux info
+	read_data = numpy.zeros(nVertex_CHTMarker_PHYS)
+	write_data = numpy.zeros(nVertex_CHTMarker_PHYS)
+
+	# Retrieve some control parameters from the driver
+	deltaT = SU2Driver.GetUnsteady_TimeStep()
+	TimeIter = SU2Driver.GetTime_Iter()
+	nTimeIter = SU2Driver.GetnTimeIter()
+	time = TimeIter*deltaT
+	
+
+	# Set up initial data for preCICE
+	if (participant.requires_initial_data()):
+
+		for i, iVertex in enumerate(iVertices_CHTMarker_PHYS):
+			write_data[i] = GetInitialFxn(CHTMarkerID, iVertex)
+
+		participant.write_data(mesh_name, precice_write, vertex_ids, write_data)
+
+	# Initialize preCICE
+	participant.initialize()
+
+	# Sleep briefly to allow for data initialization to be processed
+	# This should only be needed on some systems and use cases
+	#
+	sleep(3)
+
+	# Time loop is defined in Python so that we have access to SU2 functionalities at each time step
+	if rank == 0:
+		print("\n------------------------------ Begin Solver -----------------------------\n")
+	sys.stdout.flush()
+	if options.with_MPI == True:
+		comm.Barrier()
+
+	precice_saved_time = 0
+	precice_saved_iter = 0
+	
+	# Patch for su2code/SU2#2353
+	if (TimeIter == 0):
+		SU2Driver.Output(TimeIter)
+		TimeIter += 1
+		time += deltaT
```

Comment:
Is https://github.com/su2code/SU2/issues/2353 now converged to this fix, or is more discussion going on?
Maybe link to https://github.com/precice/su2-adapter/issues/42 in the comment, since this provides context for the adapter?